### PR TITLE
register custom_op_symbolic for squeeze

### PR DESCRIPTION
**Pull request by mindest**
**Description**: When training, export `torch.squeeze` as `Squeeze` instead of `Shape`+`Gather`+`Equal`+`If`.
image

**Motivation and Context**
- Why is this change required? What problem does it solve?
Current [`_infer_If`](https://github.com/microsoft/onnxruntime/blob/2da82fd0b9a80d012521760091d5d7a503d86730/onnxruntime/python/tools/symbolic_shape_infer.py#L940-L963) does not correctly infer shapes from then- and else- branches: `cond` (output of `Equal`) is always `[False]` when comparing constant 1 with a symbolic dim. In a 1P model, it causes failure in later shape inference of `Conv`. Removing `If` will also support gradient of `Squeeze`.

### CODE REVIEW

**Review by baijumeswani — APPROVED** [review]

**Review by iK1D — APPROVED** [review]

File: orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py [hunk 1]
```diff
@@ -4867,3 +4867,32 @@ def random_state_equal(a, b):
     assert random_state_equal(ori_random_states, new_random_states)
 
     del os.environ['ORTMODULE_FALLBACK_RETRY']
+
+
+def test_squeeze_custom_symbolic_registry():
+    class SqueezeModel(torch.nn.Module):
+        def __init__(self):
+            super(SqueezeModel, self).__init__()
+            self.conv = torch.nn.Conv2d(in_channels=3, out_channels=1024, kernel_size=14, stride=14, bias=False)
```
Comment (xadupre) [inline orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py:4876]: Dimensions seems big. Is it possible to reduce them to have a faster unit test?

**Review by xadupre — APPROVED** [review]

**Review by baijumeswani — APPROVED** [review]
